flytekit/common/tasks/presto_task.py

```
@@ -0,0 +1,144 @@
+            schema=None,
+            task_inputs=None,
+            interruptible=False,
+            discoverable=False,
+            discovery_version=None,
+            retries=1,
+            timeout=None,
+    ):
+        """
+        :param Text query: Presto query specification
+        :param flytekit.common.types.schema.Schema output_schema: Schema that represents that data queried from Presto
+        :param Text routing_group: The routing group that a Presto query should be sent to for the given environment
+        :param Text catalog: The catalog to set for the given Presto query
+        :param Text schema: The schema to set for the given Presto query
+        :param dict[Text,flytekit.common.types.base_sdk_types.FlyteSdkType] task_inputs: Optional inputs to the Presto task
+        :param bool discoverable:
+        :param Text discovery_version: String describing the version for task discovery purposes
+        :param int retries: Number of retries to attempt
+        :param datetime.timedelta timeout:
+        """
+
+        # Set as class fields which are used down below to configure implicit
+        # parameters
+        self._routing_group = routing_group or ""
+        self._catalog = catalog or ""
+        self._schema = schema or ""
+        self._inputs = task_inputs
+
+        metadata = _task_model.TaskMetadata(
+            discoverable,
+            # This needs to have the proper version reflected in it
+            _task_model.RuntimeMetadata(
+                _task_model.RuntimeMetadata.RuntimeType.FLYTE_SDK, "1.0.0",
+                "python"),
+            timeout or _datetime.timedelta(seconds=0),
+            _literals.RetryStrategy(retries),
+            interruptible,
+            discovery_version,
+            "This is deprecated!"
+        )
+
+        presto_query = _presto_models.PrestoQuery(
+            routing_group=routing_group or "",
+            catalog=catalog or "",
+            schema=schema or "",
```

i don't think schema can ever be an empty string. we should raise an exception if not provided

---

Makes sense. I'll add  a check earlier in the code